drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -1067,6 +1118,31 @@ static int mcux_flexcomm_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static uint32_t usart_intenset;
+static int mcux_flexcomm_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct mcux_flexcomm_config *config = dev->config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+		usart_intenset = USART_GetEnabledInterrupts(config->base);
+		return 0;
+	case PM_DEVICE_ACTION_TURN_ON:
+		mcux_flexcomm_init(dev);
```

Comment:
`pm_device_driver_init()` must be added to the end of `mcux_flexcomm_init()`, and will call TURN_ON, potentially followed by resume. Once `pm_device_driver_init()` is added, calling init here will result in infinite recursion.

the `mcux_flexcomm_init()` should only be called once, on init, and later reinits are a result of calls to `mcux_flexcomm_pm_action()`

Reply:
@bjarki-andreasen, thank you. I have made the change to call `pm_device_driver_init()`